### PR TITLE
itr_chip drivers: dt_driver resources for interrupts bindings

### PR DESCRIPTION
**Pull request by etienne-lms**
2 patches extracted from https://github.com/OP-TEE/optee_os/pull/5954:

"core: dt_driver: define interrupt controller drivers identifier":
@jenswi-linaro review tag applied (https://github.com/OP-TEE/optee_os/pull/5954#issuecomment-1570403576).
 
"core: dt_driver: add helper for old fashion interrupt bindings"
@jenswi-linaro review tag (https://github.com/OP-TEE/optee_os/pull/5954#pullrequestreview-1464333713) is NOT YET applied as I squashed in a fix (https://github.com/OP-TEE/optee_os/pull/5954#discussion_r1219294494).

### CODE REVIEW

File: core/kernel/dt_driver.c [hunk 1]
```diff
@@ -292,6 +292,47 @@ void *dt_driver_device_from_parent(const void *fdt, int nodeoffset,
...
+	}
+
+	prv = dt_driver_get_provider_by_phandle(phandle, type);
+	if (!prv) {
+		*res = TEE_ERROR_DEFER_DRIVER_INIT;
+		return NULL;
+	}
+
+	prop_index *= dt_driver_provider_cells(prv);
+	if (prop_index * sizeof(*prop) >= (size_t)len) {
```
Comment (jenswi-linaro) [inline core/kernel/dt_driver.c:327]: Woulnd't
``` C
(prop_index + 1) * sizeof(*prop) > (size_t)len
```
be more accurate?

Comment (etienne-lms) [reply core/kernel/dt_driver.c:327]: you're right. Thanks.

**Comment by etienne-lms** [comment]
comment addressed

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
fixup commit squashed and tag applied.
thanks.

**Pull request merged**